### PR TITLE
Reintroduced semantic date ranges

### LINKED ISSUE #4079: Using filter first-day-last-month and last-day-last-month gives search_phase_execution_exception error
After some recent updates, I cannot say exactly which version as I was not aware of this error, I am getting. 

```
url: /api/search?filters=%7B%22initial_date%22%3A%7B%22from%22%3A%22first-day-last-month%22%2C%22to%22%3A%22last-day-last-month%22%7D%7D&limit=0&order=desc&sort=creationDate&types=%5B%226cb66d666eaf666bc6666bd3%22%5D&allAggregations=true&geolocation=true&include=%5B%22permissions%22%5D
query: {
 "filters": {
  "initial_date": {
   "from": "first-day-last-month",
   "to": "last-day-last-month"
  }
 },
 "limit": 0,
 "order": "desc",
 "sort": "creationDate",
 "types": [
  "6cb66d666eaf666bc6666bd3"
 ],
 "allAggregations": true,
 "geolocation": true,
 "include": [
  "permissions"
 ]
}
search_phase_execution_exception
```

The `inital_date` is a type date field.

It was used in a `<Database/>` tag, if I replace `initial_date.from` and `inital_date.to` with proper timestamp, it works normally.

### PR DESCRIPTION
**Pull request by RafaPolit**
fixes #4079

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

File: app/api/search/specs/search.spec.js [hunk 1]
```diff
@@ -375,6 +376,34 @@ describe('search', () => {
...
+              relationshipdate: { from: 'first-day-last-month', to: 'last-day-last-month' },
+            },
+          },
+          'en'
+        );
+
+        expect(results.rows[0].title).toBe('Inherited 1 EN');
+      } catch (err) {
+        fail(err);
+      }
```
Comment (daneryl) [inline app/api/search/specs/search.spec.js:404]: @RafaPolit try catch block here is not needed, if something fails jest will report the error properly.

File: app/api/search/specs/search.spec.js [hunk 2]
```diff
@@ -872,11 +901,11 @@ describe('search', () => {
     it('should filter by template', async () => {
       userFactory.mock(undefined);
       const { options } = await search.autocomplete('en', 'en');
-      expect(options.length).toBe(4);
+      expect(options.length).toBe(6);
```
Comment (daneryl) [inline app/api/search/specs/search.spec.js:904]: @RafaPolit why all tests have a new +2 length ? the number of entities on fixtures have not changed, i guess adding the titles is the cause ?

**Review by daneryl — CHANGES_REQUESTED** [review]
@RafaPolit looks good, couple of minor comments

File: app/api/search/specs/search.spec.js (hunk 1 above)
Comment (RafaPolit) [reply app/api/search/specs/search.spec.js:404]: No, it didn't, it was not reporting the error returned by the await.  It was "just" failing.  Will look again.

File: app/api/search/specs/search.spec.js (hunk 2 above)
Comment (RafaPolit) [reply app/api/search/specs/search.spec.js:904]: Yes, exactly so, and the fact that I added the language to the title.

File: app/api/search/specs/search.spec.js (hunk 1 above)
Comment (RafaPolit) [reply app/api/search/specs/search.spec.js]: I have created a tech debt with this problem and with a quick explanation:
https://github.com/huridocs/uwazi/issues/4193

**Review by daneryl — APPROVED** [review]

**Pull request merged**